types/stylelint/index.d.ts

```diff
@@ -1,6 +1,7 @@
 import type * as PostCSS from 'postcss';
 import type { GlobbyOptions } from 'globby';
 import type { cosmiconfig, TransformSync as CosmiconfigTransformSync } from 'cosmiconfig';
+import { RequiredDeep } from 'type-fest';
```

Comment:
opinion

I think it's worth adding that dependency.
I won't mind if you disagree; if so I will just have to duplicate the types.
see https://github.com/sindresorhus/type-fest/commit/c2bf37491077ee5fbb580d913ba0c80152d65ce0

Reply:
If using `type-fest`, will there be any effect on projects depending on Stylelint's public types?

Reply:
I am not sure I understand, `RequiredDeep` is just a type provided by a dependency.
It could easily be vendored on our side and it would work the same.

Reply:
When I try this PR as a dependency in a sample project, I get the following error: ⬇️ 

```sh-session
$ npx tsc
node_modules/stylelint/types/stylelint/index.d.ts:4:35 - error TS2307: Cannot find module 'type-fest' or its corresponding type declarations.

4 import type { RequiredDeep } from 'type-fest';
                                    ~~~~~~~~~~~


Found 1 error in node_modules/stylelint/types/stylelint/index.d.ts:4
```

Maybe, it's necessary to have `type-fest` in `dependencies`, not `devDependencies`.

Reply:
However, I don't think adding `type-fest` to `dependencies` is worth just for `RequiredDeep`... 🤔

Reply:
> However, I don't think adding type-fest to dependencies is worth just for RequiredDeep... 🤔

I will remove it.

Reply:
[33a0bc6](https://github.com/stylelint/stylelint/commit/33a0bc61a67845b2cd819e15885ece4b231cb68e)